Compilation fails on CentOS7

```
make[2]: Entering directory `/__w/libvmi/libvmi/libkvmi/src'
/bin/sh ../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I..    -I../include -I../include/linux/x86_64 -g -O2 -W -Wall -MT kvmi.lo -MD -MP -MF .deps/kvmi.Tpo -c -o kvmi.lo kvmi.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -I../include -I../include/linux/x86_64 -g -O2 -W -Wall -MT kvmi.lo -MD -MP -MF .deps/kvmi.Tpo -c kvmi.c  -fPIC -DPIC -o .libs/kvmi.o
kvmi.c: In function 'handshake_done':
kvmi.c:625:9: warning: missing initializer for field 'struct_size' of 'struct kvmi_introspector2qemu' [-Wmissing-field-initializers]
  struct kvmi_introspector2qemu  intro = {};
         ^
In file included from kvmi.c:41:0:
../include/libkvmi.h:59:11: note: 'struct_size' declared here
  uint32_t struct_size;
           ^
kvmi.c: In function 'kvmi_domain_close':
kvmi.c:1161:2: error: 'for' loop initial declarations are only allowed in C99 mode
  for ( struct kvmi_dom_event *ev = dom->events; ev; ) {
  ^
kvmi.c:1161:2: note: use option -std=c99 or -std=gnu99 to compile your code
kvmi.c: In function 'kvmi_set_xsave':
kvmi.c:2561:9: warning: missing initializer for field 'id' of 'struct kvmi_msg_hdr' [-Wmissing-field-initializers]
  struct kvmi_msg_hdr  hdr      = {};
         ^
In file included from ../include/libkvmi.h:25:0,
                 from kvmi.c:41:
../include/linux/kvmi.h:92:8: note: 'id' declared here
  __u16 id;
        ^
kvmi.c:2562:9: warning: missing initializer for field 'vcpu' of 'struct kvmi_vcpu_hdr' [-Wmissing-field-initializers]
  struct kvmi_vcpu_hdr vcpu_hdr = {};
         ^
In file included from ../include/libkvmi.h:25:0,
                 from kvmi.c:41:
../include/linux/kvmi.h:204:8: note: 'vcpu' declared here
  __u16 vcpu;
        ^
make[2]: *** [kvmi.lo] Error 1
make[2]: Leaving directory `/__w/libvmi/libvmi/libkvmi/src'

--- a/examples/hookguest.c
+++ b/examples/hookguest.c
@@ -84,7 +84,9 @@ static void handle_cr_event( void *dom, struct kvmi_dom_event *ev )
 		struct kvmi_vcpu_hdr       hdr;
 		struct kvmi_event_reply    common;
 		struct kvmi_event_cr_reply cr;
-	} rpl = { 0 };
+	} rpl;
+
+	memset( &rpl, 0, sizeof( rpl ) );
 
 	printf( "CR%d 0x%llx -> 0x%llx (vcpu%u)\n", cr->cr, cr->old_value, cr->new_value, ev->event.common.vcpu );
 
@@ -99,7 +101,9 @@ static void handle_msr_event( void *dom, struct kvmi_dom_event *ev )
 		struct kvmi_vcpu_hdr        hdr;
 		struct kvmi_event_reply     common;
 		struct kvmi_event_msr_reply msr;
-	} rpl = { 0 };
+	} rpl;
+
+	memset( &rpl, 0, sizeof( rpl ) );
 
 	printf( "MSR 0x%x 0x%llx -> 0x%llx (vcpu%u)\n", msr->msr, msr->old_value, msr->new_value,
 	        ev->event.common.vcpu );
@@ -142,7 +146,7 @@ static void handle_pause_vcpu_event( void *dom, struct kvmi_dom_event *ev )
 	struct {
 		struct kvmi_vcpu_hdr    hdr;
 		struct kvmi_event_reply common;
-	} rpl             = { 0 };
+	} rpl;
 	unsigned int vcpu = ev->event.common.vcpu;
 	static bool  events_enabled[MAX_VCPU];
 
@@ -152,6 +156,8 @@ static void handle_pause_vcpu_event( void *dom, struct kvmi_dom_event *ev )
 		enable_vcpu_events( dom, vcpu );
 		events_enabled[vcpu] = true;
 	}
+
+	memset( &rpl, 0, sizeof( rpl ) );
 
 	reply_continue( dom, ev, &rpl.hdr, sizeof( rpl ) );
 }
@@ -175,13 +181,14 @@ static void maybe_start_pf_test( void *dom, struct kvmi_dom_event *ev )
 	__u64       cr3  = ev->event.common.arch.sregs.cr3;
 	__u16       vcpu = ev->event.common.vcpu;
 	__u64       pt   = cr3 & ~0xfff;
+	__u64       end;
 
 	if ( started || !pt )
 		return;
 
 	printf( "Starting #PF test with CR3 0x%llx (vcpu%u)\n", cr3, vcpu );
 
-	for ( __u64 end = pt + EPT_TEST_PAGES * PAGE_SIZE; pt < end; pt += PAGE_SIZE )
+	for ( end = pt + EPT_TEST_PAGES * PAGE_SIZE; pt < end; pt += PAGE_SIZE )
 		write_protect_page( dom, pt );
 
 	started = true;
@@ -204,8 +211,10 @@ static void handle_pf_event( void *dom, struct kvmi_dom_event *ev )
 		struct kvmi_vcpu_hdr       hdr;
 		struct kvmi_event_reply    common;
 		struct kvmi_event_pf_reply pf;
-	} rpl       = {};
+	} rpl;
 	__u8 access = KVMI_PAGE_ACCESS_R | KVMI_PAGE_ACCESS_W | KVMI_PAGE_ACCESS_X;
+
+	memset( &rpl, 0, sizeof( rpl ) );
 
 	printf( "PF gva 0x%llx gpa 0x%llx access %s [0x%x] (vcpu%u)\n", pf->gva, pf->gpa, access_str[pf->access & 7],
 	        pf->access, vcpu );

--- a/src/kvmi.c
+++ b/src/kvmi.c
@@ -622,7 +622,7 @@ static int read_qemu_data( struct kvmi_dom *dom, struct kvmi_qemu2introspector *
 static bool handshake_done( struct kvmi_ctx *ctx, struct kvmi_dom *dom )
 {
 	struct kvmi_qemu2introspector *qemu  = &dom->hsk;
-	struct kvmi_introspector2qemu  intro = {};
+	struct kvmi_introspector2qemu  intro;
 	struct iovec                   iov   = { .iov_base = &intro, .iov_len = sizeof( intro ) };
 
 	if ( read_qemu_data( dom, qemu ) ) {
@@ -630,6 +630,7 @@ static bool handshake_done( struct kvmi_ctx *ctx, struct kvmi_dom *dom )
 		return false;
 	}
 
+	memset( &intro, 0, sizeof( intro ) );
 	intro.struct_size = sizeof( intro );
 	if ( ctx->handshake_cb && ctx->handshake_cb( qemu, &intro, ctx->cb_ctx ) < 0 )
 		return false;
@@ -1148,6 +1149,7 @@ void kvmi_close( void *_ctx )
 void kvmi_domain_close( void *d, bool do_shutdown )
 {
 	struct kvmi_dom *dom = d;
+	struct kvmi_dom_event *ev;
 
 	if ( !dom )
 		return;
@@ -1158,7 +1160,7 @@ void kvmi_domain_close( void *d, bool do_shutdown )
 		shutdown( dom->fd, SHUT_RDWR );
 	close( dom->fd );
 
-	for ( struct kvmi_dom_event *ev = dom->events; ev; ) {
+	for ( ev = dom->events; ev; ) {
 		struct kvmi_dom_event *next = ev->next;
 
 		free( ev );
@@ -2558,8 +2560,8 @@ int kvmi_get_xcr( void *dom, unsigned short vcpu, __u8 xcr, __u64 *value )
 
 int kvmi_set_xsave( void *dom, unsigned short vcpu, const void *buffer, size_t size )
 {
-	struct kvmi_msg_hdr  hdr      = {};
-	struct kvmi_vcpu_hdr vcpu_hdr = {};
+	struct kvmi_msg_hdr  hdr;
+	struct kvmi_vcpu_hdr vcpu_hdr;
 	struct iovec         iov[]    = {
                 { .iov_base = &hdr, .iov_len = sizeof( hdr ) },
                 { .iov_base = &vcpu_hdr, .iov_len = sizeof( vcpu_hdr ) },
@@ -2568,9 +2570,13 @@ int kvmi_set_xsave( void *dom, unsigned short vcpu, const void *buffer, size_t s
 	size_t n          = sizeof( iov ) / sizeof( iov[0] );
 	size_t total_size = sizeof( hdr ) + sizeof( vcpu_hdr ) + size;
 
+	memset( &hdr, 0, sizeof( hdr ) );
+
 	hdr.id   = KVMI_VCPU_SET_XSAVE;
 	hdr.seq  = new_seq();
 	hdr.size = total_size - sizeof( hdr );
+
+	memset( &vcpu_hdr, 0, sizeof( vcpu_hdr ) );
 
 	vcpu_hdr.vcpu = vcpu;
 

